includes/class-algolia-settings.php

```diff
@@ -92,31 +92,42 @@ public function get_api_key() {
 	}
 
 	/**
-	 * Get the post types blacklist.
+	 * Get the excluded post types.
 	 *
 	 * @author  WebDevStudios <contact@webdevstudios.com>
 	 * @since   1.0.0
 	 *
 	 * @return array
 	 */
-	public function get_post_types_blacklist() {
-		$blacklist = (array) apply_filters( 'algolia_post_types_blacklist', array( 'nav_menu_item' ) );
+	public function get_excluded_post_types() {
+
+		// Default array of excluded post types.
+		$excluded = array( 'nav_menu_item' );		
+
+		if ( has_filter( 'algolia_excluded_post_types' ) ) {
```

Comment:
The `if ( has_filter() ) {}` checks aren't necessary.

Additionally, we should warns devs that they are using a deprecated filter that will be removed at some point. We should replace `apply_filters()` with `apply_filters_deprecated()` on those filters we are deprecating. 

Example (also might need some work)...
```
/**
 * Get the excluded post types.
 *
 * @author WebDevStudios <contact@webdevstudios.com>
 * @since  1.7.0
 *
 * @return array
 */
public function get_excluded_post_types() {

	// Default array of excluded post types.
	$excluded = [ 'nav_menu_item' ];

	/**
	 * Filters excluded post types.
	 *
	 * @since      1.0.0
	 * @deprecated 1.7.0 Use {@see 'algolia_excluded_post_types'} instead.
	 *
	 * @param array $excluded The excluded post types.
	 */
	$excluded = (array) apply_filters_deprecated(
		'algolia_post_types_blacklist',
		[ $excluded ],
		'1.7.0',
		'algolia_excluded_post_types',
		__( 'Optional message about the deprecation.' )
	);

	/**
	 * Filters excluded post types.
	 *
	 * @since 1.7.0
	 *
	 * @param array $allowed_options The excluded post types.
	 */
	$excluded = (array) apply_filters( 'algolia_excluded_post_types', $excluded );

	//  Additional method code...

	return array_unique( $excluded );
}

```